### DISCUSSION TITLE
Reactive transport couplings Errors

### DISCUSSION THREAD

**Discussion by robo-boluo** [idaholab/moose#27518] [Q&A Modules: Porous Flow] [answered]
Dear MOOSE Community,

I've been working with the transport reactive model located in the folder '/moose/modules/combined/examples/geochem-porous_flow' and have developed my own models, porous_flow.i and aquifer_chemistry.i.

To couple these models, I utilized 'MultiApps'. However, I encountered a 'parallel-communicated exception' error once the simulation reached t > 100,000s.

![WX20240430-232545](https://github.com/idaholab/moose/assets/59897601/f3a5f025-e8af-4aaf-a694-c9889e96ab4e)

You can access the models via the following links:
Google drive: (https://drive.google.com/drive/folders/1NrCVzYxLryaYVuvoumjBkrbedpDhcKFY?usp=sharing) 
Github:(https://github.com/robo-boluo/Module_test.git)

I would greatly appreciate any guidance or assistance with this issue.

Robo

**Comment by GiudGiud** [comment, marked as answer]
Hello

It is odd that a failed solve ended the calculation. We now rely on exceptions so that a failed solve triggers a second attempt with a smaller time step.

You can perform this manually with a checkpoint. You create a checkpoint in your first simulation, and restart from it with a smaller time step.
https://mooseframework.inl.gov/source/outputs/Checkpoint.html
https://mooseframework.inl.gov/application_usage/restart_recover.html

While I see that the nested geochemistry solve time step was properly reduced to try to find convergence, I think the time step of the main-app & multiapp iteration can be reduced as well.

@rpodgorney @1runer

Guillaume
